### ISSUE TITLE
FlexIT: Towards Flexible Semantic Image Translation	

### ISSUE THREAD

**Issue by chaerinkong** [reyllama/paper-reviews#169]
## TL; DR

![image](https://user-images.githubusercontent.com/46308407/163208112-0ec29b5c-cc83-454f-ad99-7f7b693a84d7.png)

- Leverage pretrained autoencoder (VQ-GAN), multimodal encoder  (CLIP) and perception network (VGG) for ***latent-optimization-based*** image manipulation (translation).
- Add heuristic regularizations to mitigate the *ill-defined-ness* of the problem.

## Motivation

- Most GAN-inversion based image editing methods are confined to the single domain the GAN model was trained on.
- Proposes to offer a unified flexible framework for text-driven image manipulation.

### Framework

![image](https://user-images.githubusercontent.com/46308407/163208180-6a32f698-f0bb-41dc-a570-0864d3be36da.png)

- Obtain a target point in multimodal embedding space via linear operation. (**P**)
- Encode the source image with pretrained autoencoder to obtain the initial latent.
- During optimization, update the latent vector using 3 objective terms while **keeping all the networks frozen**.

### Objective

- Embedding loss: *guides decoded output to be located at the target point in multimodal embedding space*
![image](https://user-images.githubusercontent.com/46308407/163208572-0102a175-e31b-4495-af87-9fd9143ac8e7.png)

- Perceptual Loss: *alter only the relevant part of the image while keeping as much of it constant*
![image](https://user-images.githubusercontent.com/46308407/163208906-fdd1e26c-05bf-400f-b42a-d75626b40187.png)

- Latent loss: *updated latent belongs to the decoder support **and** latent update is done in **sparse** manner. (local changes)*
![image](https://user-images.githubusercontent.com/46308407/163209155-71ec54ff-eedf-438d-a078-9b816e8cfbd9.png)

- Complete Objective
![image](https://user-images.githubusercontent.com/46308407/163209191-0a90c05b-98d0-44ee-be2b-5b45ffc74a00.png)

### Multimodal embedding

![image](https://user-images.githubusercontent.com/46308407/163209262-31ff7c7d-03dc-4407-955f-bfb809a8878b.png)

- Adopt three different architectural variants of CLIP that implement *complementary inductive bias*.
- For each branch, outputs from 8 different augmentations are fed and averaged.
- Outputs from three networks are simply concatenated.

### Metric

- SFID (simplified FID): *omits off-diagonal terms in the covariance matrix for numerical stability with small dataset*
- CSFID: *class-conditional SFID, the average of SFID for each class.*

### Experiments

![image](https://user-images.githubusercontent.com/46308407/163209608-76ba2799-36b6-42f8-8c44-1d88ccc9a1a7.png)
![image](https://user-images.githubusercontent.com/46308407/163209634-d5f378e2-a2d2-4ca7-8c79-dab509bbddda.png)
![image](https://user-images.githubusercontent.com/46308407/163209677-0623a335-3563-4441-8d5c-3351d06c90a6.png)
![image](https://user-images.githubusercontent.com/46308407/163209712-802bb370-4765-4c85-9da1-7d5780389ed1.png)
![image](https://user-images.githubusercontent.com/46308407/163209755-0421c16a-8381-4a80-be40-1438d1797487.png)
![image](https://user-images.githubusercontent.com/46308407/163209772-c39ccfb0-2577-4f61-9f02-9fb3dfb288ff.png)
![image](https://user-images.githubusercontent.com/46308407/163209799-763843d9-321d-4078-958a-df44837276f8.png)

## Comments

- Simple approach that leverages multiple pretrained models.
- The problem statement is appealing, while the methodology itself is relatively naive and implementational (lots of design choices and tuning I guess)
- Could facilitate much further research into this practical problem of wide-domain image manipulation.
